### DISCUSSION TITLE
Auth0 Authentication failure：TypeError: Invalid URL string.

### DISCUSSION THREAD

**Discussion by snowden-fu** [lobehub/lobehub#2551] [Self Hosting] [answered]
after clicking `login with Auth0` button, there is `Server error`.

<img width="1680" alt="image" src="https://github.com/lobehub/lobe-chat/assets/24203166/e55f0043-2c61-4d1e-ae87-4e77802f2147">
I checked the log on Vercel it looks my URL config error, but which URL?

<img width="1680" alt="image" src="https://github.com/lobehub/lobe-chat/assets/24203166/345c4d7f-7259-4037-98cc-baef07ecaaed">

**Comment by cy948** [comment]
Please provide a example of your environment variable in this deployment.

**Reply by snowden-fu** [reply]
is this okay?
<img width="485" alt="image" src="https://github.com/lobehub/lobe-chat/assets/24203166/fd6e08d8-9c11-4b74-aafb-1705b5da2ca7">

**Reply by cy948** [reply, marked as answer]
try edit the `auth0 issuer`, make it started with `https://`

**Reply by snowden-fu** [reply]
I think we can improve user experience without `https`, let them directly copy/paste to it

**Reply by cy948** [reply]
> I think we can improve user experience without `https`, let them directly copy/paste to it

Thanks, we will take it into consideration.

**Reply by arvinxx** [reply]
I think a better way it to use `s3-oss/env` to add validator for the env. In this case , we can use AUTH0_ISSUER: `z.string().url().optional()`.

Better failed in build time rather than runtime.

**Comment by snowden-fu** [comment]
> I think a better way it to use `s3-oss/env` to add validator for the env. In this case , we can use AUTH0_ISSUER: `z.string().url().optional()`.
> 
> Better failed in build time rather than runtime.

are u using `Zod` for validation? Can I contribute and make a PR to this part?

**Reply by arvinxx** [reply]
@snowden-fu  Yes, here it is: https://github.com/lobehub/lobe-chat/blob/main/src/config/auth.ts#L86

Feel free to help us to improve the validation~ 😝
